runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -61,11 +63,14 @@ void incrementLapCount(Client client)
 		} while (remainingXp < 0);
 
 		lapsTillLevel = remainingXp > 0 ? (int) Math.ceil(remainingXp / course.getTotalXp()) : 0;
+		int goalRemainingXp = client.getVar(VarPlayer.AGILITY_GOAL_END) - currentExp;
```

Comment:
this should use XpTrackerService if this is going to be implemented like this. There is variable for endGoalXp (or something like that). Remove this and use that.